### PR TITLE
Fix DynamoDB errors for records with missing info in notifications

### PR DESCRIPTION
**Pull request by OferLevi85**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java [hunk 1]
```diff
@@ -101,6 +101,13 @@ public X509CertRecord getX509CertRecord(String provider, String instanceId, Stri
     }
 
     private X509CertRecord itemToX509CertRecord(Item item) {
+        boolean clientCert;
+        try {
+            clientCert = item.getBoolean(KEY_CLIENT_CERT);
```
Comment (OferLevi85) [inline servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java:106]: If clientCert attribute doesn't exist for some reason I set it to false.

File: servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java [hunk 2]
```diff
@@ -259,21 +266,25 @@ private String getPrimaryKey(final String provider, final String instanceId, fin
...
-                            .with(":lastNotifiedTimeVal", lastNotifiedTime)
-                            .withNumber(":v_yesterday", yesterday)
-                            .withString(":lastNotifiedServerVal", lastNotifiedServer));
-
-            Item updatedItem = table.updateItem(updateItemSpec).getItem();
-
-            if (isRecordUpdatedWithNotificationTimeAndServer(lastNotifiedServer, lastNotifiedTime, updatedItem)) {
-                X509CertRecord x509CertRecord = itemToX509CertRecord(updatedItem);
-                updatedRecords.add(x509CertRecord);
+            try {
```
Comment (OferLevi85) [inline servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java:269]: If we'll find a problematic record, we'll print an error and continue so it won't affect other items.

File: servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java [hunk 3]
```diff
@@ -303,19 +314,25 @@ private boolean isRecordUpdatedWithNotificationTimeAndServer(String lastNotified
...
-                        .withNumber(":v_last_notified", yesterday)
-                        .withString(":v_provider", provider));
-
-        ItemCollection<QueryOutcome> outcome = index.query(spec);
-        List<Item> items = new ArrayList<>();
-        for (Item item : outcome) {
-            items.add(item);
+        try {
+            QuerySpec spec = new QuerySpec()
+                    .withKeyConditionExpression("currentDate = :v_current_date")
```
Comment (OferLevi85) [inline servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/DynamoDBCertRecordStoreConnection.java:319]: If an error is thrown for a specific combination of provider + date, we'll print it and continue so it won't affect other records.

**Review by havetisyan — APPROVED** [review]
👍

**Pull request merged**